src/utils/contract.ts

```diff
@@ -119,4 +120,37 @@ export default class Contract extends BaseConnection {
     });
     return await response.json();
   }
+
+  private async fetchDataViaProvider(params) {
+    switch (this.providerType) {
+      case "ethers":
+        return await this.web3Provider!
+          .sendAsync('eth_call', params)
+          .then(resp => resp.json()); 
+      case "web3":
+        try {
+          return this.web3Provider?.sendAsync({
+            jsonrpc: '2.0',
+            id: 1,
+            method: 'eth_call',
+            params,
+          }, this.fetchCallback);
+        } catch(error) {
```

Comment:
Catching **any** error is a bad practice here: we can't be sure, whether `sencAsync` is just absent or there's something else wrong in our calling code. We should just check whether `sendAsync` is not undefined instead